### PR TITLE
[fix](index compaction) base compaction failed after restore indexes

### PR DESCRIPTION
**Pull request by cambyzju**
**Problem:**
after restore from other cluster, then rowsets got different index_id, and make index compaction in base compaction always failed.

**Close this pr:**
because of duplicate pr, https://github.com/apache/doris/pull/41625 already fix it：
```
            // no inverted index or index id is different from current index id
            if (tablet_index == nullptr || tablet_index->index_id() != index.index_id()) {
                is_continue = true;
                break;
            }

```

### CODE REVIEW

**Comment by cambyzju** [comment]
run buildall

**Review by lide-reed — APPROVED** [review]
LGTM

**Comment by cambyzju** [comment]
run p0

**Comment by cambyzju** [comment]
run buildall

**Comment by cambyzju** [comment]
run External Regression

File: be/src/olap/compaction.cpp [hunk 1]
```diff
@@ -708,6 +708,19 @@ Status Compaction::do_inverted_index_compaction() {
                     break;
                 }
             }
+
+            // if we restore tablets from other cluster, then we got schema.index.index_id
+            // from src cluster in be, but in fe, we will generate new index_id,
+            // different index_id make we can not find index file correctly.
+            // Here we temporarily skip index compaction.
+            // (TODO) we should use index_id in rowset schema, not tablet schema.
```
Comment (airborne12) [inline be/src/olap/compaction.cpp:716]: Need to specify which part todo this

**Comment by cambyzju** [comment]
for branch-2.1: https://github.com/apache/doris/pull/43962

**Comment by zzzxl1993** [comment]
@cambyzju Why was this PR closed on the master branch?

**Pull request closed without merging**